lib/cosmos/gui/choosers/float_chooser.rb

```diff
@@ -12,54 +12,43 @@
 require 'cosmos/gui/choosers/value_chooser'
 
 module Cosmos
-
-  class FloatChooserDoubleValidator < Qt::DoubleValidator
-    def initialize(*args)
-      super(*args)
-    end
-
-    def fixup(input)
-      begin
-        value = input.to_f
-        if value < bottom()
-          # Handle less than bottom
-          parent().setText(bottom().to_s)
-        elsif value > top()
-          # Handle greater than top
-          parent().setText(top().to_s)
-        end
-      rescue Exception => err
-        # Oh well no fixup
-      end
-    end
-  end
-
+  # Widget which creates a horizontally laid out label and editable float value.
+  # Minimum and maximum values can be specified to perform input validation.
+  # A callback can be specified which is called once the value is changed.
   class FloatChooser < ValueChooser
     # Callback for a new value entered into the text field
     attr_accessor :sel_command_callback
 
+    # @param parent (see ValueChooser#initialize)
+    # @param label_text (see ValueChooser#initialize)
+    # @param initial_value (see ValueChooser#initialize)
+    # @param minimum_value [Float] Minimum allowable value
+    # @param maximum_value [Float] Maximum allowable value
+    # @param field_width (see ValueChooser#initialize)
+    # @param fill (see ValueChooser#initialize)
     def initialize(parent, label_text, initial_value,
                    minimum_value = nil, maximum_value = nil,
                    field_width = 20, fill = false)
       super(parent, label_text, initial_value, field_width, fill)
       @minimum_value = minimum_value
       @maximum_value = maximum_value
 
-      validator = FloatChooserDoubleValidator.new(@value)
+      validator = Qt::DoubleValidator.new(@value)
```

Comment:
The one I wrote contains a modified fixup method... Are you sure the existing one works the same?